### ISSUE TITLE
Feeding the plugin with samples@Runtime

### ISSUE THREAD

**Issue by Raj123456788** [johnboiles/coremediaio-dal-minimal-example#18]
When I select the this plugin as my camera source in Teams or zoom. I want to show a dynamic picture .How can I feed the plugin with sample buffer? Please point me to an example or documentation.

**Comment by gxalpha** [comment]
https://github.com/johnboiles/obs-mac-virtualcam uses this, you can try to look there.

**Comment by Raj123456788** [comment]
How do I debug virtualcam in Xcode?

**Comment by gxalpha** [comment]
Check out this: https://github.com/johnboiles/obs-mac-virtualcam/wiki/Developing#debugging-the-dal-plugin
It is for the obs virtualcam but should be the same for this sample.

**Comment by Raj123456788** [comment]
Details

Could not attach to pid : “9781”
Domain: IDEDebugSessionErrorDomain
Code: 3
Failure Reason: Error 1
User Info: {
    DVTRadarComponentKey = 855031;
    RawLLDBErrorMessage = "Error 1";
}
--


System Information

macOS Version 10.15.4 (Build 19E266)
Xcode 11.4 (16134)

**Comment by Raj123456788** [comment]
Can you please make the obscamplugin as xcode project?It will be easier to debug.

**Comment by sivrish** [comment]
@gxalpha I've successfully used https://github.com/johnboiles/obs-mac-virtualcam to create a virtual cam plugin but when I tried to add the Mach Server to my mac app and try to run the mach server is not starting. I'm getting the error " Unable to open mach server port. ". I followed the same as it that obs-mac-virtualcam plugin. Do I have to add any permissions to my .plist for the mach server to work or It doesn't work when it's added to the mac app itself ? Thanks in advance

**Comment by gxalpha** [comment]
Hey @sivrish ,
I'm unfortunately not completely into the details, so I'm afraid I can't help you there.
Maybe try asking @johnboiles for help, he probably has more knowledge and experience with the topic.

**Comment by sivrish** [comment]
@johnboiles can you please shed some light on this

**Comment by Raj123456788** [comment]
@sivrish: Are you able to send any buffers to the plugin? Would be great if you have an example?

**Comment by gxalpha** [comment]
@Raj123456788 the https://github.com/johnboiles/obs-mac-virtualcam uses this as the DAL plugin, you can check the functionality there if you need examples

**Comment by Raj123456788** [comment]
@gxalpha : It has been more than 10 days and I am still struggling with sending buffers. I am able to see obs sending the buffers to virtualcam. Is there a easy example that I can follow rather than trying to debug from obs-studio code repository?

**Comment by gxalpha** [comment]
No, I am not aware of any other example the the obs-mac-virtualcam.

**Comment by johnboiles** [comment]
Other than what OBS uses, you can also read about how to use XPC on Mac to send data between processes.
